tools/provision/darwin.sh

```diff
@@ -12,7 +12,7 @@ DARWIN_SETUP="\
 if [[ ! -f /var/.osquery_build ]]; then \
 touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress; \
 PROD=\$(softwareupdate -l | grep \"\\*.*Command Line\" | \
-  head -n 1 | awk -F\"*\" '{print \$2}' | sed -e 's/^ *//' | tr -d '\n' \
```

Comment:
Should we try to support `10.12` here by either `head`ing or `tail`ing according to version. Or are there other problems with compile on `10.12` that would make this irrelevant?

Reply:
This change select the most-recent version of CLI tools available vs. the oldest. I am not sure of the effect, my worst-case imagination is the old ones maintained compiler compatibility (and we were using these).